test/test-locally.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/sh
+
+oneTimeSetUp(){
+  docker network create  --opt encrypted -d overlay backend
+  docker stack deploy -c docker-compose.yml mongo
+  docker run --name client --network=backend -d mongo:3.2 tail -f /dev/null
+  echo "Created MongoDB Client"
+  eval $(docker-machine env swarm-manager)
+}
+
+# Start a new MongoDB cluster with persistence data and checks that the controller configure it correctly.
+
+testSetUpMongoIsRunning(){
+  sleep 120
+  result=$(docker service ls -f name=mongo_mongo --format "{{.Name}}:{{.Mode}}")
+  assertEquals "mongo_mongo:global" "${result}"
+}
+
+testSetUpControllerIsRunning(){
+  result=$(docker service ls -f name=mongo_controller --format "{{.Name}}:{{.Mode}}")
+  assertEquals "mongo_controller:replicated" "${result}"
+}
+
+testSetUpMongoReplicas(){
+  result=$(docker service ls -f name=mongo_mongo --format "{{.Name}}:{{.Replicas}}")
+  assertEquals "mongo_mongo:3/3" "${result}"
+}
+
+testSetUpControllerReplicas(){
+  result=$(docker service ls -f name=mongo_controller --format "{{.Name}}:{{.Replicas}}")
+  assertEquals "mongo_controller:1/1" "${result}"
+}
+
+testSetUpMongoClusterStatus(){
+  result=$(docker exec client mongo --quiet mongo_mongo/admin --eval "db.runCommand( { replSetGetStatus : 1 } )['ok']")
+  assertEquals "1" "${result}"
+}
+
+testSetUpMongoClusterSize(){
+  result=$(docker exec client mongo --quiet mongo_mongo/admin --eval "db.runCommand( { replSetGetStatus : 1 } )['members'].length")
+  assertEquals "3" "${result}"
+}
+
+# Kill container
+
+testKillMongo(){
```

Comment:
The are some like this one which are only testing docker behavior. I'd get rid of those.

All in all, is it me or there is some implicit "dependency" among these tests? If so, that'd not be good.

OK, re-reading everything, it looks to me some which are defined as tests are actually prerequisites for the actual tests located a bit below. If that's the case, I'd merge those together in a single test.

Reply:
it is surely the case (and this is to save time in the testing to re-set every time the environment). given the priorities, for the time being i would leave this unchanged. please add it as issue.

Reply:
Ok, no prob. Considering your comment in time-saving, merging in a single test method would be worse then. If they start bothering we can transform those docker-only ones to "setup" methods to be called. No need of issue for now.